### PR TITLE
Required methods were added

### LINKED ISSUE #4: Missing methods
Currently, some useful [Built-in Set](https://docs.python.org/3.7/library/stdtypes.html#set) methods do not exist in OrderedSet, such as:

- [x] is_disjoint
- [x] is_subset
- [x] is_superset
- [x] intersection
- [x] difference
- [x] union

We need to add these functionalities as an enhancement. Please see [this page](https://www.programiz.com/python-programming/methods/set) for some examples.

### PR DESCRIPTION
**Pull request by thelilnix**
Hey @buyalsky, according to #4
`OrderedSet` missed some **built-in** methods.
Now I've added them to `OrderedSet` class (also with their tests).
- [x] is_disjoint
- [x] is_subset
- [x] is_superset
- [x] intersection
- [x] difference
- [x] union

### CODE REVIEW

File: src/ordered_hash_set.py [hunk 1]
```diff
@@ -131,6 +131,109 @@ def contains_any(self, *items):
...
+
+        :param other: (list | set | OrderedSet), Checked object.
+        :return: (Bool) If the set is a subset of ``other`` return ``True``.
+        """
+        
+        for element in self:
+            if element in other:
+                continue
+            else:
+                return False
```
Comment (buyalsky) [inline src/ordered_hash_set.py]: We can simplify this as follows:
```python
if element not in other:
    return False
````

File: src/ordered_hash_set.py [hunk 2]
```diff
@@ -131,6 +131,109 @@ def contains_any(self, *items):
...
+
+        return True
+
+    def intersection(self, *other):
+        """
+        Return a new ordered set with elements common to the set and all others.
+        
+        :parm *other: (list | set | OrderedSet), The sets to check common.
```
Comment (buyalsky) [inline src/ordered_hash_set.py]: Typo: `parm`

File: src/ordered_hash_set.py [hunk 3]
```diff
@@ -131,6 +131,109 @@ def contains_any(self, *items):
...
+            else:
+                new_ordered_set.add(element)
+
+        return new_ordered_set
+
+    def difference(self, *other):
+        """
+        Return a new set with elements in the set that are not in the others.
+
+        :parm *other: (list | set | OrderedSet), The sets to check difference.
```
Comment (buyalsky) [inline src/ordered_hash_set.py]: typo: `parm`

File: src/ordered_hash_set.py [hunk 4]
```diff
@@ -131,6 +131,109 @@ def contains_any(self, *items):
...
+            else:
+                new_ordered_set.add(element)
+
+        return new_ordered_set
+
+    def union(self, *other):
+        """
+        Return a new set with elements from the set and all others.
+
+        :parm *other: (list | set | OrderedSet), The sets for union.
```
Comment (buyalsky) [inline src/ordered_hash_set.py]: typo `parm`

File: src/ordered_hash_set.py [hunk 5]
```diff
@@ -131,6 +131,109 @@ def contains_any(self, *items):
...
+
+        :param other: (list | set | OrderedSet), Checked object.
+        :return: (Bool) If the set is the superset of ``other`` return ``True``.
+        """
+        
+        for element in other:
+            if element in self:
+                continue
+            else:
+                return False
```
Comment (buyalsky) [inline src/ordered_hash_set.py]: We can simplify these lines as well.

File: src/ordered_hash_set.py [hunk 6]
```diff
@@ -131,6 +131,109 @@ def contains_any(self, *items):
...
+        Return a new ordered set with elements common to the set and all others.
+        
+        :parm *other: (list | set | OrderedSet), The sets to check common.
+        :return: (OrderedSet) the set with elements common to the OrderedSet object and all ``*other``.
+        """
+        new_ordered_set = OrderedSet()
+
+        for element in self:
+            for obj in other:
+                if not element in obj:
```
Comment (buyalsky) [inline src/ordered_hash_set.py]: According to PEP 8, non-membership test should be `if element not in obj`

File: src/ordered_hash_set.py [hunk 7]
```diff
@@ -131,6 +131,109 @@ def contains_any(self, *items):
         """
         return any(item in self for item in items)
 
+    def is_disjoint(self, other):
+        """
+        Return ``True`` if the set has no elements in common with ``other``.
+        Sets(also OrderedSets) are disjoint if and only if their intersection is the empty set.
+
+        :param other: (list | set | OrderedSet), Checked object.
+        :return: (Bool) If the set has no elements in common with other return True.
```
Comment (buyalsky) [inline src/ordered_hash_set.py]: All Python's built-in types are named as lower case, including `bool`.

File: src/ordered_hash_set.py [hunk 8]
```diff
@@ -131,6 +131,109 @@ def contains_any(self, *items):
...
+                return False
+
+        return True
+
+    def is_subset(self, other):
+        """
+        Test whether every element in the set is in other.
+
+        :param other: (list | set | OrderedSet), Checked object.
+        :return: (Bool) If the set is a subset of ``other`` return ``True``.
```
Comment (buyalsky) [inline src/ordered_hash_set.py]: As I mentioned above, we should name as `bool` instead of `Bool`.

File: src/ordered_hash_set.py [hunk 9]
```diff
@@ -131,6 +131,109 @@ def contains_any(self, *items):
...
+                return False
+
+        return True
+
+    def is_superset(self, other):
+        """
+        Test whether every element in other is in the set.
+
+        :param other: (list | set | OrderedSet), Checked object.
+        :return: (Bool) If the set is the superset of ``other`` return ``True``.
```
Comment (buyalsky) [inline src/ordered_hash_set.py]: `Bool` -> `bool`

**Review by buyalsky — COMMENTED** [review]
The PR is pretty good. I left some small comments before merging this.
On the docstrings, we should specify the function definition like `Returns a new ...` instead of `Return a new ...`

**Comment by thelilnix** [comment]
Dear @buyalsky, I'm so sorry about these mistakes.
I fixed them and I hope my PR could be merged.
Thanks for your comments.

**Comment by buyalsky** [comment]
Nice @ebsa491! Looks like the PR is ready to be merged. Thanks for the contribution.
> Dear @buyalsky, I'm so sorry about these mistakes.

No problem at all :)

**Pull request merged**